apps/vscode/src/app/tree-view/current-workspace-tree-provider.ts

```diff
@@ -0,0 +1,61 @@
+import { WorkspaceDefinition } from '@angular-console/server';
+import { existsSync, readFileSync } from 'fs';
+import { join } from 'path';
+import { TreeDataProvider } from 'vscode';
+
+import {
+  WorkspaceRoute,
+  WorkspaceRouteTitle
+} from '../tree-item/workspace-route';
+import { AbstractTreeProvider } from './abstract-tree-provider';
+
+export class CurrentWorkspaceTreeProvider extends AbstractTreeProvider<
+  WorkspaceRoute
+> {
+  static create(
+    workspacePath: string | undefined,
+    extensionPath: string
+  ): TreeDataProvider<WorkspaceRoute> {
+    return new CurrentWorkspaceTreeProvider(
+      workspacePath && existsSync(join(workspacePath, 'angular.json'))
+        ? {
+            path: workspacePath,
+            name: JSON.parse(
+              readFileSync(join(workspacePath, 'angular.json'), 'utf8')
```

Comment:
add a catch around it, so if json is not parseable, the extension still shows something

even better, maybe try to parse out the name without parsing json

Reply:
I added a try catch block. I'm not confident enough in my regexp to make a reliable way to strip the name without parsing.